crates/oxc_ecmascript/src/side_effects/context.rs

```diff
@@ -2,6 +2,18 @@ use oxc_ast::ast::Expression;
 
 use crate::is_global_reference::IsGlobalReference;
 
+#[derive(Default, Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PropertyReadSideEffects {
+    /// Treat all property read accesses as side effect free.
+    None,
+    /// Treat non-member property accesses as side effect free.
+    /// Member property accesses are still considered to have side effects.
+    OnlyMemberPropertyAccess,
+    /// Treat all property read accesses as possible side effects.
+    #[default]
+    All,
+}
```

Comment:
Rollup's default is `OnlyMemberPropertyAccess` which ignores side effects for `{ bar } = foo`. I set the default as `All` here so it's safer.